### PR TITLE
Fix city districts and increase default size

### PR DESCRIPTION
**Pull request by NotFuji**
#### Summary
```SUMMARY: Infrastructure "Fix city districts and increase default size"```  

#### Purpose of change
Cities were spawning with virtually no houses due to truncation errors, so that has been fixed.  This dropped the number of commercial buildings by about half, so the default city size has been doubled.

#### Describe the solution
`town_dist` was calculated as a decimal percentage of the radius of the town and stored as an int, making it effectively always 0.  This has been converted into an integer percentage.

The random selection of building type has been converted to normal distribution for more natural district boundries.

Residential:Commercial ratio changed to more accurately reflect reality, city size is increased to compensate for the subsequent lack of commercial buildings.

#### Additional context
Before (default size 4):

After (default size 8):

After (max size 16):

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
What about region settings json? Would you put sigma and new radius values there?

File: src/overmap.cpp [hunk 1]
```diff
@@ -2332,9 +2332,21 @@ void overmap::place_cities()
...
-    if( rng( 0, 99 ) > settings.city_spec.shop_radius * town_dist ) {
+    int shop_radius = settings.city_spec.shop_radius;
+    int park_radius = settings.city_spec.park_radius;
+
+    int shop_sigma = settings.city_spec.shop_sigma;
+    int park_sigma = settings.city_spec.park_sigma;
+
+    //Normally distribute shops and parks
+    //Clamp at 1/2 radius to prevent houses from spawning in the city center.
+    //Parks are nearly guarenteed to have a non-zero chance of spawning anywhere in the city.
```
Comment (ZhilkinSerg) [inline src/overmap.cpp]:
```suggestion
    //Parks are nearly guaranteed to have a non-zero chance of spawning anywhere in the city.
```

**Comment by chaisawlajatang** [comment]
Oh God yes, yesssss! Towns would finally look like places where people actually LIVE.

**Comment by NotFuji** [comment]
> The presence of extensive number of 4x2 blocks in the big city seems quite awkward and unnatural. Shouldn't they be mixed with longer sections of 5,6,7,8 x 2 also?

I didn't change that at all in this PR, they currently spawn as 3-5x2

**Comment by ZhilkinSerg** [comment]
Yeah, they are just became visible because houses are spawned.

**Comment by NotFuji** [comment]
The new values are in regional_map_settings.json now.  I was wondering why the spawnrates were acting weird.

**Comment by ZhilkinSerg** [comment]
`\data\json\test_regions.json` too please.

**Comment by ZhilkinSerg** [comment]
I agree city mapgen should be eventually changed. There was stalled attempt at city blocks - https://github.com/CleverRaven/Cataclysm-DDA/pull/22186.

**Pull request merged**